circleci: failed tests (14133): TestClusterRecovery

The following test appears to have failed:

[#14133](https://circleci.com/gh/cockroachdb/cockroach/14133):

```
I160309 18:29:07.486135 acceptance/chaos_test.go:178  client 2 shutting down
I160309 18:29:07.487760 acceptance/chaos_test.go:178  client 0 shutting down
I160309 18:29:07.513134 acceptance/chaos_test.go:361  40 7.3/sec
W160309 18:29:07.700835 acceptance/chaos_test.go:250  round 3: cluster recovered
I160309 18:29:07.801244 acceptance/cluster/localcluster.go:585  stopping
--- FAIL: TestClusterRecovery (11.69s)
	chaos_test.go:268: pq: storage/stores.go:87: store 2 not found
=== RUN   TestNodeRestart
I160309 18:29:09.799293 acceptance/cluster/localcluster.go:254  Initializing Cluster:
Cluster Setup: AdHoc 3x1
Duration: 5s, Stall: 2m0s
Node 0 - Store 0
Node 1 - Store 0
Node 2 - Store 0
Total Nodes Count:3, Total Store Count:3
I160309 18:29:09.799481 acceptance/cluster/docker.go:73  ImagePull cockroachdb/builder:20160305-182433 starting
--
--- SKIP: TestBuildBabyCluster (0.00s)
	terraform_test.go:30: only enabled during testing
=== RUN   TestFiveNodesAndWriters
--- SKIP: TestFiveNodesAndWriters (0.00s)
	util_test.go:64: running in docker mode
FAIL
ok github.com/cockroachdb/cockroach/acceptance 1337s

```
Please assign, take a look and update the issue accordingly.

--- a/server/server.go
+++ b/server/server.go
@@ -295,10 +295,6 @@ func (s *Server) Start() error {
 		}))
 	})
 
-	s.stopper.RunWorker(func() {
-		util.FatalIfUnexpected(m.Serve())
-	})
-
 	s.gossip.Start(s.grpc, unresolvedAddr)
 
 	// Register admin service
@@ -330,6 +326,10 @@ func (s *Server) Start() error {
 
 	log.Infof("starting %s server at %s", s.ctx.HTTPRequestScheme(), unresolvedHTTPAddr)
 	log.Infof("starting grpc/postgres server at %s", unresolvedAddr)
+
+	s.stopper.RunWorker(func() {
+		util.FatalIfUnexpected(m.Serve())
+	})
 
 	return nil
 }

